### PR TITLE
Runtime hunting: Chem mask on the floor edition

### PR DESCRIPTION
**Pull request by DamianX**
```
x10 chemmask.dm,290: undefined variable /turf/simulated/shuttle/floor/var/back
  proc name: pack check (/obj/item/clothing/mask/chemmask/proc/pack_check)
  src: the chemical mask (/obj/item/clothing/mask/chemmask)
  src.loc: the floor (147,93,2) (/turf/simulated/shuttle/floor)
  call stack:
  the chemical mask (/obj/item/clothing/mask/chemmask): pack check(the floor (147,93,2) (/turf/simulated/shuttle/floor))
  the chemical mask (/obj/item/clothing/mask/chemmask): process()
  Objects (/datum/subsystem/obj): fire(0)
  Objects (/datum/subsystem/obj): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

### CODE REVIEW

**Comment by jknpj** [comment]
this is fucking hilarious

**Comment by Probe1** [comment]
😨

File: code/modules/clothing/masks/chemmask.dm [hunk 1]
```diff
@@ -287,7 +287,7 @@
 
 /obj/item/clothing/mask/chemmask/proc/pack_check(mob/user) //Shuts off mask if the user is not wearing a chempack.
 	var/mob/living/M = user
-	if (!(M && M.back && istype(M.back,/obj/item/weapon/reagent_containers/chempack)))
+	if (!(istype(M) && M.back && istype(M.back,/obj/item/weapon/reagent_containers/chempack)))
 		mask_shutdown(user)
```
Comment (Shadowmech88) [inline code/modules/clothing/masks/chemmask.dm:291]: The code would still get here if the `istype(M)` fails, and proceed to attempt to call `update_inv_wear_mask()` on a non-mob in `mask_shutdown()`. An `istype()` check should be added to that proc as well.

**Review by Shadowmech88 — CHANGES_REQUESTED** [review]

**Comment by DamianX** [comment]
Tested it too, for good measure.

**Review by Shadowmech88 — APPROVED** [review]

**Pull request merged**